src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -276,6 +276,24 @@ public void Remove(object key)
             StartScanForExpiredItemsIfNeeded(_options.Clock.UtcNow);
         }
 
+        /// <summary>
+        /// Removes all keys and values from the cache.
+        /// </summary>
+        public void Clear()
+        {
+            CheckDisposed();
+
+            var oldEntries = Interlocked.Exchange(ref _entries, new ConcurrentDictionary<object, CacheEntry>());
+            Interlocked.Exchange(ref _cacheSize, 0);
+
+            foreach (var entry in oldEntries)
+            {
+                entry.Value.SetExpired(EvictionReason.Removed);
+                entry.Value.InvokeEvictionCallbacks();
+            }
+            oldEntries.Clear();
```

Comment:
Why is this oldEntries.Clear line needed?

Reply:
It's not needed (GC will cleanup the memory anyway)